src/hooks/useFetch.ts

```diff
@@ -0,0 +1,97 @@
+import { useEffect, useReducer, useRef } from "react";
```

Comment:
Please write unit test for this file

Reply:
Actually... why do you need something like this, when you use it exactly once? It seems like another way of fetching data in the dashboard, so once you introduce something like this, everyone is supposed to follow this, while I don't think we need this, especially when you want to make just one API call that is not graphql

Reply:
Agree, as a rule of thumb -> 1 usage keep in component file, if you keep duplicating -> extract

Reply:
I agree we should keep things duplicated first and only when needed refactor them, but without generic fetch hook we still need logic to differentiate loading, data and error to do it well, so it still maybe done in separate hook, thus I don't see big difference in this particular case.

Reply:
It's mostly clone of https://usehooks-ts.com/react-hook/use-fetch, which I see as just a handy react helper hook for browser fetch interface. I can obviously add tests for it.

Reply:
I've moved it into `src/new-apps/hooks`, changed from `useFetch` to `useMarketplaceApps` and tested it.